### PR TITLE
Modify how we create the dataset so it doesn't timeout

### LINKED ISSUE #1887: Quantpendia command timmed out for MUS_MUSCULUS
Error:

```
ubuntu@ip-10-0-127-71:~$ docker logs inspiring_goldberg
2019-11-11 22:18:45,252 i-04bb28b499152d24f [volume: -1] data_refinery_foreman.foreman.management.commands.create_quantpendia INFO [organism: HOMO_SAPIENS] [job_id: 29380081]: Sending compendia job for Organism
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
psycopg2.extensions.QueryCanceledError: canceling statement due to statement timeout


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 23, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/home/user/data_refinery_foreman/foreman/management/commands/create_quantpendia.py", line 83, in handle
    job = create_job_for_organism(organism)
  File "/home/user/data_refinery_foreman/foreman/management/commands/create_quantpendia.py", line 34, in create_job_for_organism
    data[experiment.accession_code] = list(samples_with_quantsf)
  File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 268, in __iter__
    self._fetch_all()
  File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 1186, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 176, in __iter__
    for row in compiler.results_iter(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size):
  File "/usr/local/lib/python3.5/dist-packages/django/db/models/sql/compiler.py", line 1017, in results_iter
    results = self.execute_sql(MULTI, chunked_fetch=chunked_fetch, chunk_size=chunk_size)
  File "/usr/local/lib/python3.5/dist-packages/django/db/models/sql/compiler.py", line 1065, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.5/dist-packages/raven/contrib/django/client.py", line 127, in execute
    return real_execute(self, sql, params)
  File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.5/dist-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.OperationalError: canceling statement due to statement timeout
```

Worked for human.

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

close #1887 

## Purpose/Implementation Notes

Split the query to calculate the quantpendia dataset, so that it doesn't timeout. 

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

Tested function in prod for `MUS_MUSCULUS`

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py [hunk 1]
```diff
@@ -72,10 +91,10 @@ def handle(self, *args, **options):
         logger.debug('Generating quantpendia for organisms', organisms=all_organisms)
 
         for organism in all_organisms:
-            # only generate the quantpendia for organisms that have some samples
-            # with quant.sf files.
+            # only generate the quantpendia for organisms that at least one
+            # RNA-SEQ processed sample
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: What's the reason for this change? This is significantly altering the existing behavior.

File: foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py [hunk 2]
```diff
@@ -53,6 +36,42 @@ def create_job_for_organism(organism: Organism):
...
+                .filter(is_processed=True)\
+                .values_list('accession_code', flat=True)
+            # and only check for quant file for unprocessed samples
+            unprocessed_samples_with_quantsf = experiment_samples\
+                .filter(
+                    is_processed=False,
+                    results__computedfile__filename='quant.sf'
+                )\
+                .values_list('accession_code', flat=True)\
+                .distinct()
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: Both of these sub-queries are no longer filtering on the organism. Any reason why not?

Comment (arielsvn) [reply foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: good catch will fix it

File: foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py (hunk 1 above)
Comment (arielsvn) [reply foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: I thought they were equivalent and I wanted some alternative to avoid doing `.filter(results__computedfile__filename='quant.sf')` which is very expensive for the DB.

This will exclude any organisms that have `quant.sf` files but don't have any processed RNA-SEQ samples. Currently, no organism in our DB meets this condition. but if we were to have one, do we want a quantpendia for them? Note that people can download `quant.sf` files through the smasher as well.

```py
>>> Organism.objects.filter(qn_target__isnull=False).exclude(sample__is_processed=True, sample__technology='RNA-SEQ').filter(sample__results__computedfile__filename='quant.sf').exists()
False
```

Comment (kurtwheeler) [reply foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: Hmm, not too sure, but I guess my gut is "if we can, why not?"

Comment (arielsvn) [reply foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: Good point reverted the condition.

File: foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py [hunk 3]
```diff
@@ -91,10 +91,10 @@ def handle(self, *args, **options):
         logger.debug('Generating quantpendia for organisms', organisms=all_organisms)
 
         for organism in all_organisms:
-            # only generate the quantpendia for organisms that at least one
-            # RNA-SEQ processed sample
+            # only generate the quantpendia for organisms that have some samples
+            # with quant.sf files.
             has_quantsf_files = organism.sample_set\
-                .filter(is_processed=True, technology='RNA-SEQ')\
+                .filter(results__computedfile__filename='quant.sf')\
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: Wouldn't
> technology='RNA-SEQ'
still be a good filter to keep?

Comment (arielsvn) [reply foreman/data_refinery_foreman/foreman/management/commands/create_quantpendia.py]: sure, good optimization!

```suggestion
                .filter(technology='RNA-SEQ', results__computedfile__filename='quant.sf')\
```
